src/Cryptol/TypeCheck/AST.hs

```diff
@@ -532,10 +532,14 @@ instance PP n => PP (WithNames (ModuleG n)) where
          , vcat (map pp' mDecls)
 
          , vcat (map pp (Map.elems mFunctors))
+
+         , vcat (map ppSig (Map.toList mSignatures))
```

Comment:
...calling `vcat` here when there are no `mSignatures` will result in an extra newline being printed. Does `prettyprinter` offer a combinator that will collapse the newlines to avoid this problem?